include/kernel.h

```diff
@@ -181,6 +181,32 @@ struct _k_object {
 	u32_t data;
 } __packed;
 
+struct _k_object_assignment {
+	struct k_thread *thread;
+	void * const *objects;
+};
+
+/**
+ * @brief Grant a static thread access to a list of kernel objects
+ *
+ * For threads declared with K_THREAD_DEFINE(), grant the thread access to
+ * a set of kernel objects. These objects do not need to be in an initialized
+ * state. The permissions will be granted when the threads are initialized
+ * in the early boot sequence.
+ *
+ * All arguments beyond the first should be pointers to kernel objects.
```

Comment:
"should be" or "must be"? What's the consequence if they're pointers to non-kernel objects?  (Same question for the other "should be" mentions below (so I won't repeat).